crates/reward/src/lib.rs

```diff
@@ -278,22 +278,26 @@ impl<T: Config> Pallet<T> {
     }
 }
 
-pub trait Rewards<VaultId> {
+pub trait Rewards<VaultId, CurrencyId> {
     /// Signed fixed point type.
     type SignedFixedPoint: FixedPointNumber;
 
     /// Return the stake associated with the `account_id`.
-    fn get_stake(account_id: &VaultId) -> Self::SignedFixedPoint;
+    fn get_stake(account_id: &VaultId, currency_id: CurrencyId) -> Self::SignedFixedPoint;
 
     /// Deposit an `amount` of stake to the `account_id`.
     fn deposit_stake(account_id: &VaultId, amount: Self::SignedFixedPoint) -> Result<(), DispatchError>;
 
     /// Distribute the `reward` to all participants OR return the leftover.
-    fn distribute_reward(reward: Self::SignedFixedPoint) -> Result<Self::SignedFixedPoint, DispatchError>;
+    fn distribute_reward(
```

Comment:
Please introduce the following error state: https://github.com/interlay/interbtc/pull/286/files#diff-63acba1a1c0ed1a214390cea807f4b5f3e17506e74aa37617bae05d6c1950f45R203-R206

Reply:
In my version there is no `GetRewardCurrencyIds`. I could throw this error when it is neither the vault's backing currency nor `GetNativeCurrencyId`, what do you think?

Reply:
Yeah I think that still makes sense to have

Reply:
Actually on second look, this isn't possible, since `distribute_reward` and `get_total_rewards` do not take vault_ids. I think that is ok though since imo this error check isn't really necessary - the code will continue to function as expected anyway